cli/pkg/kctrl/cmd/package/available/generate-default-values.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package available
+
+import (
+	"bytes"
+	"reflect"
+
+	"gopkg.in/yaml.v3"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	structuralschema "k8s.io/apiextensions-apiserver/pkg/apiserver/schema"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const Object = "object"
+
+// SchemaDefault returns a yaml byte array with values populated according to schema
+func SchemaDefault(schema []byte) ([]byte, error) {
+	jsonSchemaProps := &apiextensions.JSONSchemaProps{}
+
+	if err := yaml.Unmarshal(schema, jsonSchemaProps); err != nil {
+		return nil, err
+	}
+	s, err := structuralschema.NewStructural(jsonSchemaProps)
+	if err != nil {
+		return nil, err
+	}
+	unstructured := make(map[string]interface{})
+
+	schemaDefault(unstructured, s)
+	var b bytes.Buffer
+	yamlEncoder := yaml.NewEncoder(&b)
+	yamlEncoder.SetIndent(2)
+	if err := yamlEncoder.Encode(unstructured); err != nil {
```

Comment:
challenge with us generating "default values" file is that if user just takes that yaml and uses it as values.yml, then user effectively says "override everything" but in general user just needs to override just a few values. i believe @jorgemoralespou proposed an idea that we should generate a file that has all the fields commented. so that user can just uncomment fields and set values to what they want. additionally it would be nice if we can include descriptions on top of values and indicate what type.

Reply:
> we should generate a file that has all the fields commented. so that user can just uncomment fields and set values to what they want.

I really like the idea. 

> additionally it would be nice if we can include descriptions on top of values and indicate what type.

That would be great. Let me see if I can add that as part of the same PR (the title minor changes doesn't feel suitable now 😁 )